stacksCollectible/index.ts

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { StacksNetwork } from '@stacks/network';
+import BigNumber from 'bignumber.js';
+import { NftCollectionData, NftDetailResponse, NftEventsResponse, NonFungibleToken } from 'types';
+import { getNftDetail, getNftsCollectionData } from '../api/gamma';
+import { getNftsData } from '../api/stacks';
+import { microstacksToStx } from '../currency';
+import { getCacheValue, setCacheValue } from './stacksCollectionCache';
+
+export interface StacksCollectionData {
+  collection_id: string | null;
+  collection_name: string | null;
+  total_nft: number;
+  thumbnail_nfts: Array<NonFungibleToken>; //stores a max of four nfts
+  all_nfts: Array<NonFungibleToken>; //stores entire list of nft in collection
+  floor_price?: number;
+}
+export interface StacksCollectionList {
+  offset: number;
+  limit: number;
+  total: number;
+  results: Array<StacksCollectionData>;
+}
+
+async function getAllNftContracts(address: string, network: StacksNetwork): Promise<NonFungibleToken[]> {
+  const listofContracts: Array<NonFungibleToken> = [];
+
+  //make initial call to get the total inscriptions count and limit
+  let offset = 0;
+  const response = await getNftsData(address, network, 0);
+  const total = response.total;
+  offset += response.limit;
+  listofContracts.push(...response.results);
+
+  let listofContractPromises: Array<Promise<NftEventsResponse>> = [];
+
+  // make API calls in parallel to speed up fetching data
+  for (let i = offset; i <= total; i += response.limit) {
+    listofContractPromises.push(getNftsData(address, network, i));
+
+    if (listofContractPromises.length === 4) {
+      const resolvedPromises = await Promise.all(listofContractPromises);
+      resolvedPromises.forEach((resolvedPromise) => listofContracts.push(...resolvedPromise.results));
+      listofContractPromises = [];
+    }
+  }
+
+  return listofContracts;
+}
+
+async function fetchBatchData(batch: Array<NonFungibleToken>, collectionRecord: Record<string, StacksCollectionData>) {
+  const nftDetailPromises: Array<Promise<NftDetailResponse>> = [];
+  const collectionDataPromises: Array<Promise<NftCollectionData | undefined>> = [];
+  const nftArray: Array<NonFungibleToken> = [];
+
+  batch.forEach((nft) => {
+    const principal: string[] = nft.asset_identifier.split('::');
+    const contractInfo: string[] = principal[0].split('.');
+    const contractId = principal[0];
```

Comment:
```suggestion
    const principal: string[] = nft.asset_identifier?.split('::');
    const contractInfo: string[] = principal[0]?.split('.');
    const contractId = principal[0];
```

nit: I would usually be more defensive when dealing with external API responses like this